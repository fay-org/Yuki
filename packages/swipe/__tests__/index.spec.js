import { mount } from '@vue/test-utils'
import Swipe from '../index'
import SwipeItem from '../../swipe-item'

describe('Swipe', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Swipe)
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should be support indicatorColor', () => {
    const wrapper = mount(
      <Swipe indicatorColor="red">
        <SwipeItem />
      </Swipe>,
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should be support custom indicatorSize', () => {
    const wrapper = mount(
      <Swipe indicatorHeight="6px" indicatorWidth="6px">
        <SwipeItem />
      </Swipe>,
    )
    expect(wrapper.html()).toMatchSnapshot()
  })
})
