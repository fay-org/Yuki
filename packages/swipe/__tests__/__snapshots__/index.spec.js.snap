// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swipe should be support custom indicatorSize 1`] = `
<div class="fect-swipe">
  <div class="fect-swipe__track" style="height: 0px; width: 0px; transform: translateX(0px); transition-duration: 300ms;">
    <div class="fect-swipe-item" style="height: 0px; width: 0px; transform: translateX(0px);">
      <!---->
    </div>
  </div>
  <!---->
</div>
`;

exports[`Swipe should be support indicatorColor 1`] = `
<div class="fect-swipe">
  <div class="fect-swipe__track" style="height: 0px; width: 0px; transform: translateX(0px); transition-duration: 300ms;">
    <div class="fect-swipe-item" style="height: 0px; width: 0px; transform: translateX(0px);">
      <!---->
    </div>
  </div>
  <!---->
</div>
`;
